### PR TITLE
SLAM: Process images when they're of YCbCr type

### PR DESCRIPTION
**Pull request by kkufieta**
Implements the RDK side of this ticket: https://viam.atlassian.net/browse/DATA-521

Tested with:
* realsense: mono & rgbd
* webcam: mono

### CODE REVIEW

File: services/slam/builtin/builtin.go [hunk 1]
```diff
@@ -735,14 +735,22 @@ func (slamSvc *builtIn) getLazyPNGImage(ctx context.Context, cam camera.Camera)
 		return nil, nil, err
 	}
 
-	lazyImg, ok := img.(*rimage.LazyEncodedImage)
-	if !ok {
-		return nil, nil, errors.Errorf("expected lazily encoded image, got %T", lazyImg)
+	if lazyImg, ok := img.(*rimage.LazyEncodedImage); ok {
```
Comment (tessavitabile) [inline services/slam/builtin/builtin.go:738]: Do I understand correctly that for realsense (mono or RGBD), we will still get a lazy PNG here, so there is no difference in the images passed to orbslam?

File: services/slam/builtin/builtin.go [hunk 2]
```diff
@@ -735,14 +735,22 @@ func (slamSvc *builtIn) getLazyPNGImage(ctx context.Context, cam camera.Camera)
...
+		if lazyImg.MIMEType() != utils.MimeTypePNG {
+			return nil, nil, errors.Errorf("expected mime type %v, got %T", utils.MimeTypePNG, img)
+		}
+		return lazyImg.RawData(), release, nil
 	}
-	if lazyImg.MIMEType() != utils.MimeTypePNG {
-		return nil, nil, errors.Errorf("expected mime type %v, got %v", utils.MimeTypePNG, lazyImg.MIMEType())
+	
+	if ycbcrImg, ok := img.(*image.YCbCr); ok {
```
Comment (tessavitabile) [inline services/slam/builtin/builtin.go:745]: I suggest adding an integration test that runs mono YCbCr images, but it's fine with me if you want to do that in a follow-up ticket.

Comment (kkufieta) [reply services/slam/builtin/builtin.go:745]: I'll create a follow-up ticket for this.

File: services/slam/builtin/builtin.go (hunk 1 above)
Comment (kkufieta) [reply services/slam/builtin/builtin.go:738]: That is correct.

File: services/slam/builtin/builtin.go [hunk 3]
```diff
@@ -735,14 +735,22 @@ func (slamSvc *builtIn) getLazyPNGImage(ctx context.Context, cam camera.Camera)
...
+		pngImage, err := rimage.EncodeImage(ctx, ycbcrImg, utils.MimeTypePNG)
+		if err != nil {
+			return nil, nil, err
+		}
+		return pngImage, release, nil
 	}
-	return lazyImg.RawData(), release, nil
+
+	return nil, nil, errors.Errorf("expected lazily encoded image or ycbcrImg, got %T", img)
 }
```
Comment (tessavitabile) [inline services/slam/builtin/builtin.go:754]: [nit] You could change the function name to `getPNGImage()`, since it will not always be lazy.

**Review by tessavitabile — APPROVED** [review]

File: services/slam/builtin/builtin.go (hunk 3 above)
Comment (kkufieta) [reply services/slam/builtin/builtin.go:754]: Added it to the follow-up ticket (create integration test): DATA-535

**Review by JohnN193 — APPROVED** [review]
LGTM, was curious about the images being BGR vs RGB with these changes+slam changes but we can look more into that later

**Pull request merged**